### PR TITLE
Alma 51  unmarshalling  of  web service result

### PR DESCRIPTION
**Pull request by gregorgabriel**
I had to change application/xml to application/json to make i unmarshall.
Tried to simplify HttpServiceException for later use.

### CODE REVIEW

File: src/main/java/no/unit/alma/commons/HttpStatusException.java [hunk 1]
```diff
@@ -1,30 +1,47 @@
...
+     * @param statusText Status message
+     * @param method HttpMethod
+     * @param url request url
+     * @param responseBody responseBody
+     * @param webServiceResult webServiceResult object from AlmaŒŒ
+     */
     public HttpStatusException(int status, String statusText, String method, String url, String responseBody,
-            WebServiceResult webServiceResult) {
+                               WebServiceResult webServiceResult) {
         super(status + " " + statusText + " " + method + " " + url);
```
Comment (Eiriknil) [inline src/main/java/no/unit/alma/commons/HttpStatusException.java]: Bruk String.format

Comment (gregorgabriel) [reply src/main/java/no/unit/alma/commons/HttpStatusException.java]: Ay Sir.

File: build.gradle [hunk 2]
```diff
@@ -174,12 +175,12 @@ pmdMain { exclude 'no/unit/alma/generated/**' }
...
     ruleSetConfig = rootProject.resources.text.fromFile('config/pmd/ruleset.xml')
     ruleSets = []
-    ignoreFailures = true
+//    ignoreFailures = true
 }
 
 checkstyle {
     configFile = rootProject.resources.text.fromFile('config/checkstyle/checkstyle.xml').asFile()
-    showViolations = true
+//    showViolations = true
```
Comment (placbo) [inline build.gradle]: why?

File: src/main/java/no/unit/alma/commons/HttpStatusException.java [hunk 3]
```diff
@@ -1,30 +1,47 @@
...
+     * The HttpStatusException is called as a filter on communication with AlmaClient in order to catch
+     * WebServiceResults, containing the error-details from Alma. The HttpServiceException will be caught be a
+     * ResponseProcessingException. Catching that, will enable you to access the almaErrorCode etc. pp..
+     *
+     * @param status Status number
+     * @param statusText Status message
+     * @param method HttpMethod
+     * @param url request url
+     * @param responseBody responseBody
+     * @param webServiceResult webServiceResult object from AlmaŒŒ
```
Comment (placbo) [inline src/main/java/no/unit/alma/commons/HttpStatusException.java]: ŒŒ

File: build.gradle (hunk 2 above)
Comment (gregorgabriel) [reply build.gradle]: oy, just a test om et gjorde noe forskjell. Skulle ikke commites.

File: src/main/java/no/unit/alma/commons/HttpStatusException.java (hunk 3 above)
Comment (gregorgabriel) [reply src/main/java/no/unit/alma/commons/HttpStatusException.java]: Min mac gir meg mye trøbbel på shortcuts, det kan jeg love deg.

**Review by placbo — APPROVED** [review]

**Review by Eiriknil — APPROVED** [review]

**Pull request merged**